client/components/NavBar/NavBar.tsx

```diff
@@ -19,81 +23,81 @@ const defaultProps = {
 
 const NavBar = function(props) {
 	const { communityData } = usePageContext(props.previewContext);
-	const pages = communityData.pages || [];
-	const navigation = communityData.navigation || [];
-	const navItems = populateNavigationIds(pages, navigation);
+	const { pages = [], collections = [], navigation = [] } = communityData;
+	const navItems = getNavItemsForCommunityNavigation({
+		navigation: navigation,
+		pages: pages,
+		collections: collections,
+	});
 	const socialItems = populateSocialItems(communityData);
+
 	return (
 		<nav className="nav-bar-component accent-background accent-color">
 			<GridWrapper>
 				<div className="scrollable-nav">
 					<ul className="nav-list">
 						{navItems
-							.filter((item) => {
-								return !!item;
-							})
+							.filter((x) => x)
 							.map((item) => {
-								/* Return Dropdown */
-								if (item.children) {
+								if ('children' in item) {
 									return (
 										// @ts-expect-error
 										<Menu
 											aria-label={item.title}
 											disclosure={
 												<li className="dropdown">
-													{item.title}
+													<span className="title">{item.title}</span>
 													<span className="bp3-icon-standard bp3-icon-caret-down bp3-align-right" />
 												</li>
 											}
 											className="nav-bar-popover"
 											key={`nav-item-${item.id}`}
 										>
 											{item.children.map((subitem, index) => {
-												return (
-													<MenuItem
-														// eslint-disable-next-line react/no-array-index-key
-														key={index}
-														// @ts-expect-error
-														href={subitem.href || `/${subitem.slug}`}
-														icon={
-															subitem.slug &&
-															!subitem.isPublic && (
-																<Icon icon="lock2" iconSize={14} />
-															)
-														}
-														rightElement={
-															subitem.href && (
-																<Icon icon="share" iconSize={14} />
-															)
-														}
-														text={subitem.title}
-													/>
-												);
+												if ('href' in subitem) {
+													return (
+														<MenuItem
+															// eslint-disable-next-line react/no-array-index-key
+															key={index}
+															// @ts-expect-error
+															href={subitem.href}
+															icon={
+																subitem.isPrivate && (
+																	<Icon
+																		icon="lock2"
+																		iconSize={14}
+																	/>
+																)
+															}
+															rightElement={
+																subitem.isExternal && (
+																	<Icon
+																		icon="share"
+																		iconSize={14}
+																	/>
+																)
+															}
+															text={subitem.title}
+														/>
+													);
+												}
+												return null;
```

Comment:
This is pretty deep indentation, at least in the GitHub diff. I wonder if we could wrap any of this in a helper function at the top-level of the component scope, or do an early return or two.

Reply:
Sure thing — I actually did this and then reverted it because it made the diff a big mess, but the point is well-taken.